plugins/package/src/main/kotlin/net/corda/cli/plugins/packaging/CreateCpb.kt

```diff
@@ -0,0 +1,129 @@
+    lateinit var outputCpbFileName: String
+
+    @Option(names = ["--keystore", "-s"], required = true, description = ["Keystore holding signing keys"])
+    lateinit var keyStoreFileName: String
+
+    @Option(names = ["--storepass", "--password", "-p"], required = true, description = ["Keystore password"])
+    lateinit var keyStorePass: String
+
+    @Option(names = ["--key", "-k"], required = true, description = ["Key alias"])
+    lateinit var keyAlias: String
+
+    @Option(names = ["--tsa", "-t"], description = ["Time Stamping Authority (TSA) URL"])
+    var tsaUrl: String? = null
+
+    companion object {
+        private val CPB_NAME_ATTRIBUTE = Attributes.Name("Corda-CPB-Name")
+
+        private val CPB_VERSION_ATTRIBUTE = Attributes.Name("Corda-CPB-Version")
+
+        private val CPB_FORMAT_VERSION = Attributes.Name("Corda-CPB-Format")
+
+        const val CPB_CURRENT_FORMAT_VERSION = "2.0"
+
+        private val CPB_UPGRADE = Attributes.Name("Corda-CPB-Upgrade")
+
+        const val CPB_FILE_EXTENSION = "cpb"
+
+        /**
+         * Name of signature within Cpb file
+         */
+        private const val CPB_SIGNER_NAME = "CPB-SIG"
+
+        /**
+         * Check file exists and returns a Path object pointing to the file, throws error if file does not exist
+         */
+        private fun requireFileExists(fileName: String): Path {
+            val path = Path.of(fileName)
+            require(Files.isReadable(path)) { "\"$fileName\" does not exist or is not readable" }
+            return path
+        }
+
+        /**
+         * Check that file does not exist and returns a Path object pointing to the filename, throws error if file exists
+         */
+        private fun requireFileDoesNotExist(fileName: String): Path {
+            val path = Path.of(fileName)
+            require(Files.notExists(path)) { "\"$fileName\" already exists" }
+            return path
+        }
+    }
+
+    override fun run() {
+        val unsignedCpb = Files.createTempFile("buildCPB", ".$CPB_FILE_EXTENSION")
+        try {
+            buildUnsignedCpb(unsignedCpb, cpks)
+            val cpbPath = requireFileDoesNotExist(outputCpbFileName)
+            val privateKeyEntry = CpxSigner.getPrivateKeyEntry(keyStoreFileName, keyStorePass, keyAlias)
+            val privateKey = privateKeyEntry.privateKey
+            val certPath = CpxSigner.buildCertPath(privateKeyEntry.certificateChain.asList())
+            CpxSigner.sign(unsignedCpb, cpbPath, privateKey, certPath, CPB_SIGNER_NAME, tsaUrl)
+        } finally {
+            Files.deleteIfExists(unsignedCpb)
+        }
+    }
+
+    private fun buildUnsignedCpb(unsignedCpb: Path, cpks: List<String>) {
+        cpks.forEach {
+            requireFileExists(it)
+        }
+
+        val manifest = Manifest()
+        val manifestMainAttributes = manifest.mainAttributes
+        manifestMainAttributes[Attributes.Name.MANIFEST_VERSION] = "1.0"
```

Comment:
Is this meant to be hard-coded?  Or something that @josephzunigadaly knows about?

Reply:
Thanks for the catch, since this is now merged will address in subsequent PR.